cmd/metal-api/internal/datastore/integer.go

```diff
@@ -15,15 +15,32 @@ var (
 	IntegerPoolRangeMax = uint(131072)
 )
 
+// IntegerPool manages unique integers
+type IntegerPool struct {
+	tablename string
+	min       uint
+	max       uint
+	rs        *RethinkStore
+}
+
 type integer struct {
 	ID uint `rethinkdb:"id" json:"id"`
 }
 
-// Integerinfo contains information on the integer pool.
-type Integerinfo struct {
+// integerinfo contains information on the integer pool.
+type integerinfo struct {
 	IsInitialized bool `rethinkdb:"isInitialized" json:"isInitialized"`
 }
 
+// GetIntegerPool returns a named integerpool if already created
+func (rs *RethinkStore) GetIntegerPool(name string) (*IntegerPool, error) {
```

Comment:
Do we really need a pool map, which requires error handling on pool-user-side? Isn't it possible to just assign both pools to two variables and to access the directly?

Reply:
i hate global variables

Reply:
typed getter which creates the pool if not already initialized